### PR TITLE
`ST_DISTANCE` function

### PR DESCRIPTION
**Pull request by JCOR11599**
TODO:
 - distance over geospatial coordinates
 - units

### CODE REVIEW

File: sql/expression/function/distance.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	}
+	return fmt.Sprintf("ST_DISTANCE(%s)", strings.Join(args, ","))
+}
+
+// WithChildren implements the Expression interface.
+func (d *Distance) WithChildren(children ...sql.Expression) (sql.Expression, error) {
+	return NewDistance(children...)
+}
+
+// FindGeometryDistance finds the minimum distance between points of any geometry types a and b
```
Comment (andrew-wm-arthur) [inline sql/expression/function/distance.go:70]: this function doesn't find minimum distance between geometries, it finds _a_ distance between two arbitrary points within each geometry

Comment (zachmu) [reply sql/expression/function/distance.go:70]: Seems like this works fine right? It's N-squared, compares all points in the first geometries with all points in the second

File: sql/expression/function/distance.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+	}
+
+	// unit must be string
+	_unit, err := sql.LongText.Convert(unit)
+	if err != nil {
+		return nil, err
+	}
+
+	// Must be a valid unit
+	switch _unit.(string) {
```
Comment (zachmu) [inline sql/expression/function/distance.go:179]: Given that there's no happy path here, I would just move this check to the constructor.

**Review by zachmu — APPROVED** [review]
LGTM

**Pull request closed without merging**